backend/fms_core/viewsets/full_sample.py

```diff
@@ -0,0 +1,184 @@
+                **(dict(coordinates=full_sample['coordinates']) if full_sample['coordinates'] is not None else dict()),
+                **(dict(concentration=full_sample['concentration']) if full_sample['concentration'] is not None else dict()),
+            )
+
+            sample = Sample.objects.create(**sample_data)
+
+            DerivedBySample.objects.create(derived_sample_id=derived_sample.id,
+                                           sample_id=sample.id,
+                                           volume_ratio=1)
+
+        except ValidationError as err:
+            raise ValidationError(err)
+
+        serializer = SampleSerializer(sample)
+        data = serializer.data
+        return Response(data)
+
+    def update(self, request, *args, **kwargs):
+        full_sample = request.data
+
+        #Retrive the sample to update
+        try:
+            sample_to_update = Sample.objects.get(pk=full_sample['id'])
+        except Exception as e:
+            raise ValidationError(e)
+
+        #Update individual fields
+        if full_sample['name']:
+            sample_to_update.name = full_sample['name']
+        if full_sample['container']:
+            try:
+                new_container = Container.objects.get(pk=full_sample['container'])
+            except Exception as e:
+                raise ValidationError(e)
+            sample_to_update.container = new_container
+        if full_sample['coordinates']:
+            sample_to_update.coordinates = full_sample['coordinates']
+        if full_sample['volume']:
+            sample_to_update.volume = full_sample['volume']
+        if full_sample['concentration']:
+            sample_to_update.concentration = float_to_decimal(full_sample['concentration'])
+        if full_sample['depleted'] is not None:
+            sample_to_update.depleted = full_sample['depleted']
+
+        #Save the new sample
+        try:
+            sample_to_update.save()
+        except ValidationError as err:
+            raise ValidationError(err)
+
+        #Return new sample
+        serializer = SampleSerializer(sample_to_update)
+        data = serializer.data
+        return Response(data)
+
+    @action(detail=False, methods=["get"])
+    def list_export(self, _request):
+        serializer = FullSampleExportSerializer(self.filter_queryset(self.get_queryset()), many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["get"])
+    def summary(self, _request):
+        """
+        Returns summary statistics about the current set of samples in the
+        database.
+        """
+        return Response({"total_count": Sample.objects.all().count()})
+
+    @action(detail=False, methods=["get"])
+    def list_collection_sites(self, _request):
+        samples_data = Biosample.objects.filter().distinct("collection_site")
+        collection_sites = [s.collection_site for s in samples_data]
+        return Response(collection_sites)
+
```

Comment:
Missing get serializer class to provide the FullSampleNestedSerializer if requested... (it's not used inside Freezeman but it is part of the "reporting" endpoint.") People currently use this endpoint from outside the application.